python/feathub/registries/mysql_registry.py

```diff
@@ -0,0 +1,289 @@
+    ).hexdigest()
+
+
+class MySqlRegistry(Registry):
+    """
+    A registry that stores entities in a MySQL database.
+    """
+
+    def __init__(
+        self,
+        props: Dict,
+    ):
+        super().__init__("mysql", props)
+        mysql_registry_config = MySqlRegistryConfig(props)
+        self.database = mysql_registry_config.get(DATABASE_CONFIG)
+        self.table = mysql_registry_config.get(TABLE_CONFIG)
+        self.host = mysql_registry_config.get(HOST_CONFIG)
+        self.port = mysql_registry_config.get(PORT_CONFIG)
+        self.username = mysql_registry_config.get(USERNAME_CONFIG)
+        self.password = mysql_registry_config.get(PASSWORD_CONFIG)
+
+        # dict that acts as a local cache for built and registered descriptors.
+        # each value in the dict is a tuple of
+        # - original descriptor.
+        # - resolved descriptor.
+        # - UTC timestamp when the descriptor is resolved.
+        self.tables: Dict[str, Tuple[TableDescriptor, TableDescriptor, datetime]] = {}
+
+        self.conn = mysql.connector.connect(
+            host=self.host,
+            port=self.port,
+            user=self.username,
+            password=self.password,
+            database=self.database,
+        )
+        self.cursor = self.conn.cursor()
+
+        self.cursor.execute(
+            f"""
+                CREATE TABLE IF NOT EXISTS `{self.table}`(
+                   `name` VARCHAR(64) NOT NULL,
+                   `timestamp` TIMESTAMP NOT NULL,
+                   `digest` VARCHAR(64) NOT NULL,
+                   `original_descriptor` TEXT NOT NULL,
+                   `resolved_descriptor` TEXT NOT NULL,
+                   PRIMARY KEY ( `name`, `timestamp` )
+                );
+            """
+        )
+
+    def build_features(
+        self,
+        feature_descriptors: List[TableDescriptor],
+        force_update: bool = False,
+        props: Optional[Dict] = None,
+    ) -> List[TableDescriptor]:
+        result = []
+        for table in feature_descriptors:
+            if table.name == "":
+                raise FeathubException(
+                    "Cannot build a TableDescriptor with empty name."
+                )
+
+            # TODO: add document about the limitations on the length of feature
+            #  table names.
+            if len(table.name) > 64:
+                raise FeathubException(
+                    "Cannot build or register a descriptor with a name longer"
+                    "than 64 characters."
+                )
+
+            self.tables[table.name] = (
+                table,
+                table.build(self, force_update=force_update, props=props),
+                datetime.utcnow(),
+            )
+            result.append(self.tables[table.name][1])
+
+        return result
+
+    def register_features(
+        self, feature_descriptors: List[TableDescriptor], force_update: bool = False
+    ) -> bool:
```

Comment:
Can you update this method to return `List[bool]` so that the caller knows which descriptors are successfully registered?